app/views/account/deletions/show.html.erb

```diff
@@ -21,6 +21,9 @@
 <ul>
   <li><%= t ".delete_profile" %></li>
   <li><%= t ".delete_display_name" %></li>
+  <% unless current_user.changesets.exists? -%>
```

Comment:
My initial thought is that we should abstract this slightly, e.g. `unless current_user.must_retain_email?` or similar. As per simonpoole's comment, there might be other situations where we need to retain the email.

Reply:
I've rebased this on the current master and introduced the `must_retain_email?` method to centralise the logic around when it should be retained as suggested.